### PR TITLE
Vagrant updates

### PR DESCRIPTION
**Pull request by tobiasd**
* bring the Vagrant VM to Debian Bullseye (stable)
* update the documentation
* clean-up the provision script

### CODE REVIEW

File: doc/Vagrant.md [hunk 1]
```diff
@@ -10,27 +10,26 @@ Getting started
...
-Inside, you'll find a "Vagrantfile" and some scripts in the utils folder.
-3. Run "vagrant up" from inside the friendica clone:
-        $> vagrant up
-Be patient: When it runs for the first time, it downloads an Ubuntu Server image.
-4. Run "vagrant ssh" to log into the virtual machine to log in to the VM:
-        $> vagrant ssh
+Inside, you'll find a `Vagrantfile` and some scripts in the `bin/dev` folder.
+3. Run `vagrant up` from inside the friendica clone.
+This will start the virtual machine.
+Be patient: When it runs for the first time, it downloads an Debian Server image and installs Friendica.
```
Comment (annando) [inline doc/Vagrant.md]: "`a` Debian Server image"

File: bin/dev/vagrant_provision.sh [hunk 2]
```diff
@@ -94,11 +94,7 @@ rm -rf /var/www/
 ln -fs /vagrant /var/www
 
 # install deps with composer
```
Comment (fabrixxm) [inline bin/dev/vagrant_provision.sh]: I would remove the comment, as composer is not used anymore

File: doc/Vagrant.md [hunk 3]
```diff
@@ -10,27 +10,26 @@ Getting started
...
 1. Install VirtualBox and vagrant.
 Please use an up-to-date vagrant version from https://www.vagrantup.com/downloads.html.
 2. Git clone your Friendica repository.
-Inside, you'll find a "Vagrantfile" and some scripts in the utils folder.
-3. Run "vagrant up" from inside the friendica clone:
-        $> vagrant up
-Be patient: When it runs for the first time, it downloads an Ubuntu Server image.
-4. Run "vagrant ssh" to log into the virtual machine to log in to the VM:
-        $> vagrant ssh
+Inside, you'll find a `Vagrantfile` and some scripts in the `bin/dev` folder.
```
Comment (fabrixxm) [inline doc/Vagrant.md:20]: As `composer install `is not run anymore in provisioning script, there should be a step between `git clone` and `vagrant up` which instructs to run `composer install`

File: bin/dev/vagrant_provision.sh [hunk 4]
```diff
@@ -120,10 +116,6 @@ bin/console user password "$ADMIN_NICK" "$ADMIN_PASSW"
 bin/console user add "$USER_NICK" "$USER_NICK" "$USER_NICK@friendica.local" en
 bin/console user password "$USER_NICK" "$USER_PASSW"
 
-# set the admin
-bin/console config config admin_email ""$ADMIN_NICK@friendica.local""
```
Comment (fabrixxm) [inline bin/dev/vagrant_provision.sh:124]: why not set `admin` user as admin anymore?

File: bin/dev/vagrant_provision.sh (hunk 2 above)
Comment (fabrixxm) [reply bin/dev/vagrant_provision.sh]: I forgot: why not running composer in provisioning script?

**Review by fabrixxm — CHANGES_REQUESTED** [review]
While you're at it, in [vagrant_provision.sh#L85](https://github.com/friendica/friendica/blob/4960d0f438002176fde5265cb3e23bb42f6777fc/bin/dev/vagrant_provision.sh#L85) the script setups aliases for users in local mail server, but the users are still the old "friendica[1-5]".. there should be aliases for "admin" and "friendica", or better for `$ADMIN_NICK` and `$USER_NICK`

File: bin/dev/vagrant_provision.sh (hunk 2 above)
Comment (tobiasd) [reply bin/dev/vagrant_provision.sh]: if there is a version mis-match between the guest and host system, it wont be written to the host file system. So doing it on the host will have it in the synced directory, while doing it in the guest system may have it on the host system.

File: bin/dev/vagrant_provision.sh (hunk 4 above)
Comment (tobiasd) [reply bin/dev/vagrant_provision.sh:124]: When you use the predefined config file, the admin is set there. Setting it here again will lead to an error message in the Vagrant console.

File: bin/dev/vagrant_provision.sh (hunk 2 above)
Comment (fabrixxm) [reply bin/dev/vagrant_provision.sh]: Ok, so it should be documented to be run in docs (as in other comment)

**Comment by tobiasd** [comment]
I've set the correct aliases during the provision.

**Review by fabrixxm — APPROVED** [review]

**Pull request merged**